### PR TITLE
luci-app-acme: convert to client side rendering

### PR DESCRIPTION
**Pull request by zhanhb**

### CODE REVIEW

**Comment by feckert** [comment]
Could you please define the map / section / option that [way](https://github.com/openwrt/luci/blob/master/applications/luci-app-samba4/htdocs/luci-static/resources/view/samba4.js#L19) ?
That is common

File: applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            _("Key size (and type) for the generated certificate."));
+        kl.value("2048", "RSA 2048 bits");
+        kl.value("3072", "RSA 3072 bits");
+        kl.value("4096", "RSA 4096 bits");
+        kl.value("ec-256", "ECC 256 bits");
+        kl.value("ec-384", "ECC 384 bits");
+        kl.default = "2048";
+        kl.rmempty = false;
+
+        if (uhttpd_presence) {
```
Comment (feckert) [inline applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js:65]: Also use the following check https://github.com/openwrt/luci/blob/master/applications/luci-app-samba4/htdocs/luci-static/resources/view/samba4.js#L51 to decide if we use uhttps or nginx

File: applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        e.rmempty = false;
+
+        var us = cs.option(form.Flag, "use_staging", _("Use staging server"),
+            _("Get certificate from the Letsencrypt staging server " +
+                "(use for testing; the certificate won't be valid)."));
+        us.rmempty = false;
+
+        var kl = cs.option(form.ListValue, "keylength", _("Key size"),
+            _("Key size (and type) for the generated certificate."));
+        kl.value("2048", "RSA 2048 bits");
```
Comment (feckert) [inline applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js:57]: I am not sure but this strings should also get translatet with _()

File: applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+        var st = s.option(form.Value, "state_dir", _("State directory"),
+            _("Where certs and other state files are kept."));
+        st.rmempty = false;
+        st.datatype = "directory";
+
+        var ae = s.option(form.Value, "account_email", _("Account email"),
+            _("Email address to associate with account key."))
+        ae.rmempty = false;
+        ae.datatype = "minlength(1)";
```
Comment (feckert) [inline applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js:38]: Is this the right datatype?

Comment (zhanhb) [reply applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js]: Here a real email address is required here. I cannot find the appropriate datatype from the document. Type `"minlength(1)"` is copied from the original lua source.

**Comment by feckert** [comment]
Sorry I have a small remark again!
Can you please switch from spaces to tabs?

**Comment by zhanhb** [comment]
Done

**Comment by feckert** [comment]
@tohojo could please verify that.
You are registered as maitainer.

@zhanhb From my side LGTM

**Comment by feckert** [comment]
As I said before, from my point of view the port looks good. So I'm going to merge this now.

Thanks

**Pull request merged**